webapp/components/FilterPosts/FilterPostsMenuItems.vue

```diff
@@ -0,0 +1,126 @@
+<template>
+  <ds-container>
+    <ds-space />
+    <ds-flex id="filter-posts-header">
+      <ds-heading tag="h4">{{ $t('filter-posts.header') }}</ds-heading>
+      <ds-space margin-bottom="large" />
+    </ds-flex>
+    <ds-flex>
+      <ds-flex-item
+        :width="{ base: '100%', sm: '100%', md: '100%', lg: '5%' }"
+        class="categories-menu-item"
+      >
+        <ds-flex>
+          <ds-flex-item width="10%" />
+          <ds-flex-item width="100%">
+            <ds-button
+              icon="check"
+              @click.stop.prevent="toggleCategory()"
+              :primary="allCategories"
+            />
+            <ds-flex-item>
+              <label class="category-labels">{{ $t('filter-posts.all') }}</label>
+            </ds-flex-item>
+            <ds-space />
+          </ds-flex-item>
+        </ds-flex>
+      </ds-flex-item>
+      <ds-flex-item :width="{ base: '0%', sm: '0%', md: '0%', lg: '4%' }" />
+      <ds-flex-item
+        :width="{ base: '0%', sm: '0%', md: '0%', lg: '3%' }"
+        id="categories-menu-divider"
+      />
+      <ds-flex-item
+        :width="{ base: '50%', sm: '50%', md: '50%', lg: '11%' }"
+        v-for="index in chunk.length"
+        :key="index"
+      >
+        <ds-flex v-for="category in chunk[index - 1]" :key="category.id" class="categories-menu">
+          <ds-flex class="categories-menu">
+            <ds-flex-item width="100%" class="categories-menu-item">
+              <ds-button
+                :icon="category.icon"
+                :primary="isActive(category.id)"
+                @click.stop.prevent="toggleCategory(category.id)"
+              />
+              <ds-space margin-bottom="small" />
+            </ds-flex-item>
+            <ds-flex>
+              <ds-flex-item class="categories-menu-item">
+                <label class="category-labels">{{ category.name }}</label>
+              </ds-flex-item>
+              <ds-space margin-bottom="xx-large" />
+            </ds-flex>
+          </ds-flex>
+        </ds-flex>
+      </ds-flex-item>
+    </ds-flex>
+  </ds-container>
+</template>
+<script>
+export default {
+  props: {
+    chunk: { type: Array, default: () => [] },
+  },
+  data() {
+    return {
+      selectedCategoryIds: [],
+      allCategories: true,
+    }
+  },
+  methods: {
+    isActive(id) {
+      const index = this.selectedCategoryIds.indexOf(id)
+      if (index > -1) {
+        return true
+      }
+      return false
+    },
+    toggleCategory(id) {
+      if (!id) {
+        this.selectedCategoryIds = []
+        this.allCategories = true
+      } else {
+        const index = this.selectedCategoryIds.indexOf(id)
+        if (index > -1) {
+          this.selectedCategoryIds.splice(index, 1)
+        } else {
+          this.selectedCategoryIds.push(id)
+        }
+        this.allCategories = false
+      }
+      this.$emit('filterPosts', this.selectedCategoryIds)
+    },
+  },
+}
+</script>
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-10T22:10:04Z" title="Thursday, July 11th 2019, 12:10:04 am +02:00">Jul 11, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,221 @@
+<template>
+  <ds-container>
+    <ds-space />
+    <ds-flex>
+      <ds-flex-item
+        :width="{ base: '100%', sm: '100%', md: '100%', lg: '20%' }"
+        class="categories-list"
+      >
+        <ds-space margin-bottom="x-small" />
+        <ds-flex id="filter-posts-header">
+          <ds-heading tag="h4">{{ $t('filter-posts.header') }}</ds-heading>
+          <ds-flex-item width="10%" />
+          <ds-flex-item width="100%">
+            <ds-button
+              icon="check"
+              @click.stop.prevent="toggleCategory()"
+              :primary="allCategories"
+            />
+            <ds-flex-item>
+              <label>{{ $t('filter-posts.all') }}</label>
+            </ds-flex-item>
+            <ds-space />
+          </ds-flex-item>
+        </ds-flex>
+      </ds-flex-item>
+      <ds-flex-item :width="{ base: '50%', sm: '0%', md: '50%', lg: '10%' }">
+        <ds-flex v-for="category in chunk[0]" :key="category.id" class="categories-list">
+          <ds-flex>
+            <ds-flex-item width="100%">
+              <ds-button
+                :icon="category.icon"
+                :primary="isActive(category.id)"
+                @click.stop.prevent="toggleCategory(category.id)"
+              />
+              <ds-space margin-bottom="small" />
+            </ds-flex-item>
+            <ds-flex>
+              <ds-flex-item>
+                <label>{{ category.name }}</label>
+              </ds-flex-item>
+              <ds-space margin-bottom="xx-large" />
+            </ds-flex>
+          </ds-flex>
+        </ds-flex>
+      </ds-flex-item>
+      <ds-flex-item :width="{ base: '50%', sm: '0%', md: '50%', lg: '10%' }">
+        <ds-flex v-for="category in chunk[1]" :key="category.id" class="categories-list">
+          <ds-flex>
+            <ds-flex-item width="100%">
+              <ds-button
+                :icon="category.icon"
+                :primary="isActive(category.id)"
+                @click.stop.prevent="toggleCategory(category.id)"
+              />
+              <ds-space margin-bottom="small" />
+            </ds-flex-item>
+            <ds-flex class="categories-list">
+              <ds-flex-item>
+                <label>{{ category.name }}</label>
+              </ds-flex-item>
+              <ds-space margin-bottom="xx-large" />
+            </ds-flex>
+          </ds-flex>
+        </ds-flex>
+      </ds-flex-item>
+      <ds-flex-item :width="{ base: '50%', sm: '0%', md: '50%', lg: '10%' }">
+        <ds-flex v-for="category in chunk[2]" :key="category.id" class="categories-list">
+          <ds-flex>
+            <ds-flex-item width="100%">
+              <ds-button
+                :icon="category.icon"
+                :primary="isActive(category.id)"
+                @click.stop.prevent="toggleCategory(category.id)"
+              />
+              <ds-space margin-bottom="small" />
+            </ds-flex-item>
+            <ds-flex>
+              <ds-flex-item>
+                <label>{{ category.name }}</label>
+              </ds-flex-item>
+              <ds-space margin-bottom="xx-large" />
+            </ds-flex>
+          </ds-flex>
+        </ds-flex>
+      </ds-flex-item>
+      <ds-flex-item :width="{ base: '50%', sm: '0%', md: '50%', lg: '10%' }">
+        <ds-flex v-for="category in chunk[3]" :key="category.id" class="categories-list">
+          <ds-flex>
+            <ds-flex-item width="100%">
+              <ds-button
+                :icon="category.icon"
+                :primary="isActive(category.id)"
+                @click.stop.prevent="toggleCategory(category.id)"
+              />
+              <ds-space margin-bottom="small" />
+            </ds-flex-item>
+            <ds-flex>
+              <ds-flex-item>
+                <label>{{ category.name }}</label>
+              </ds-flex-item>
+              <ds-space margin-bottom="xx-large" />
+            </ds-flex>
+          </ds-flex>
+        </ds-flex>
+      </ds-flex-item>
+      <ds-flex-item :width="{ base: '50%', sm: '0%', md: '50%', lg: '10%' }">
+        <ds-flex v-for="category in chunk[4]" :key="category.id" class="categories-list">
+          <ds-flex>
+            <ds-flex-item width="100%">
+              <ds-button
+                :icon="category.icon"
+                :primary="isActive(category.id)"
+                @click.stop.prevent="toggleCategory(category.id)"
+              />
+              <ds-space margin-bottom="small" />
+            </ds-flex-item>
+            <ds-flex>
+              <ds-flex-item>
+                <label>{{ category.name }}</label>
+              </ds-flex-item>
+              <ds-space margin-bottom="xx-large" />
+            </ds-flex>
+          </ds-flex>
+        </ds-flex>
+      </ds-flex-item>
+      <ds-flex-item :width="{ base: '50%', sm: '0%', md: '50%', lg: '10%' }">
+        <ds-flex v-for="category in chunk[5]" :key="category.id" class="categories-list">
+          <ds-flex>
+            <ds-flex-item width="100%">
+              <ds-button
+                :icon="category.icon"
+                :primary="isActive(category.id)"
+                @click.stop.prevent="toggleCategory(category.id)"
+              />
+              <ds-space margin-bottom="small" />
+            </ds-flex-item>
+            <ds-flex>
+              <ds-flex-item>
+                <label>{{ category.name }}</label>
+              </ds-flex-item>
+              <ds-space margin-bottom="xx-large" />
+            </ds-flex>
+          </ds-flex>
+        </ds-flex>
+      </ds-flex-item>
+      <ds-flex-item :width="{ base: '50%', sm: '0%', md: '50%', lg: '10%' }">
+        <ds-flex v-for="category in chunk[6]" :key="category.id" class="categories-list">
+          <ds-flex>
+            <ds-flex-item width="100%">
+              <ds-button
+                :icon="category.icon"
+                :primary="isActive(category.id)"
+                @click.stop.prevent="toggleCategory(category.id)"
+              />
+              <ds-space margin-bottom="small" />
+            </ds-flex-item>
+            <ds-flex>
+              <ds-flex-item>
+                <label>{{ category.name }}</label>
+              </ds-flex-item>
+              <ds-space margin-bottom="xx-large" />
+            </ds-flex>
+          </ds-flex>
+        </ds-flex>
+      </ds-flex-item>
+      <ds-flex-item :width="{ base: '50%', sm: '0%', md: '50%', lg: '10%' }">
+        <ds-flex v-for="category in chunk[7]" :key="category.id" class="categories-list">
+          <ds-flex>
+            <ds-flex-item width="100%">
+              <ds-button
+                :icon="category.icon"
+                :primary="isActive(category.id)"
+                @click.stop.prevent="toggleCategory(category.id)"
+              />
+              <ds-space margin-bottom="small" />
+            </ds-flex-item>
+            <ds-flex>
+              <ds-flex-item>
+                <label>{{ category.name }}</label>
+              </ds-flex-item>
+              <ds-space margin-bottom="xx-large" />
+            </ds-flex>
+          </ds-flex>
+        </ds-flex>
+      </ds-flex-item>
+    </ds-flex>
+  </ds-container>
+</template>
+<script>
+export default {
+  props: {
+    chunk: { type: Array, default: () => [] },
+  },
+  data() {
+    return {
+      selectedCategoryIds: [],
+      allCategories: true,
+    }
+  },
+  methods: {
+    isActive(id) {
+      const index = this.selectedCategoryIds.indexOf(id)
+      if (index > -1) {
+        return true
+      }
+      return false
+    },
+    toggleCategory(id) {
+      const index = this.selectedCategoryIds.indexOf(id)
+      if (index > -1) {
+        this.selectedCategoryIds.splice(index, 1)
+      } else {
+        this.selectedCategoryIds.push(id)
+      }
+      if (!id) this.selectedCategoryIds = []
+      this.selectedCategoryIds.length ? (this.allCategories = false) : (this.allCategories = true)
+      this.$emit('filterPosts', this.selectedCategoryIds)
+    },
+  },
+}
+</script>
```


maybe this is an improvement @roschaefer @Tirokk @ulfgebhardt @ogerly 
I think it might actually add lines of code, but it extracts the menu items into their own component